galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/routes/RunQueryRoute.java

```diff
@@ -368,21 +373,18 @@ public List<RasRunResult> sortData(List<RasRunResult> runs, QueryParameters quer
 		return runs;
 	}
 
-	public String extractSortValue (QueryParameters params) throws InternalServletException {
-		return params.getSingleString("sort","to:desc");
-	}
 
-	public Instant getWorkingFromValue (QueryParameters params) throws InternalServletException{
+	public Instant getWorkingFromValue (RasQueryParameters params) throws InternalServletException{
 		int querysize = params.getSize();
 		Instant from = null ;
 		if (querysize > 0){
-			if (!params.checkAtLeastOneQueryParameterPresent("from", "runname")){
+			if (!params.checkFromTimeOrRunNamePresent()){
 				//  RULE: Throw exception because a query exists but no from date has been supplied
 				// EXCEPT: When a runname is present in the query
 					ServletError error = new ServletError(GAL5010_FROM_DATE_IS_REQUIRED);
 					throw new InternalServletException(error, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
 			}
-			from = params.getSingleInstant("from", null);
+			from = params.getFromTime();
 		}else {
 			// The default for 'from' is now-24 hours. If no query parameters are specified
 			from = Instant.now().minus(24,ChronoUnit.HOURS);
```

Comment:
Using this Instant.now() raw like this makes it hard to unit test, and know what the 'from' is going to be. Seems we should wrap now() in a TimeService interface or something, then a mock can always return the same value if required. May not be important but it's the type of thing which causes unit tests to fail sometimes but not at other times as the value of Instant.now() will vary. Just a thought for consideration. The TimeService could be injected in at the top perhaps.

Reply:
moved the instant default outside the function to enable accruate unit testing. I don't think there is a benefit to creating a time service at this moment in time due to the fact that only one query is dependant on it